app/src/main/java/com/realityexpander/tasky/domain/validation/validateEmail/EmailMatcherRegexImpl.kt

```diff
@@ -0,0 +1,10 @@
+package com.realityexpander.tasky.domain.validation.validateEmail
+
+// Matches how the email is validated on the server
+class EmailMatcherRegexImpl: IEmailMatcher {
+    override fun matches(email: String): Boolean {
+        // use regex to validate email
+        val regex = Regex("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", RegexOption.IGNORE_CASE)
+        return regex.matches(email)
+    }
+}
```

Comment:
In this case you could also put this matching logic in the email use case directly, since there's nothing Android specific about that which you need to abstract here.

Reply:
OK, because the Android-specific implementation does not match the implementation on the server. 

Which is why I had to include this Regex version.

I will remove the Android-specific implementation.